Manager/src/main/java/io/raspberrywallet/manager/Manager.java

```diff
@@ -49,34 +47,39 @@
     @NotNull
     private final Database database;
 
-    private int autoLockRemainingMinutes = 10;
-
-    public Manager(@NotNull Database database,
-                   @NotNull List<Module> modules,
-                   @NotNull Bitcoin bitcoin,
-                   @NotNull TemperatureMonitor tempMonitor) {
+    @NotNull
+    private final CommunicationChannel frontendChannel;
+    private final AutoLockTimer autoLockTask;
+
+    Manager(@NotNull Configuration configuration,
+            @NotNull Database database,
+            @NotNull List<Module> modules,
+            @NotNull Bitcoin bitcoin,
+            @NotNull TemperatureMonitor tempMonitor,
+            @NotNull CommunicationChannel frontendChannel) {
+        this.frontendChannel = frontendChannel;
         modules.forEach(module -> this.modules.put(module.getId(), module));
         this.database = database;
         this.bitcoin = bitcoin;
         this.tempMonitor = tempMonitor;
         this.wpaConfiguration = new WPAConfiguration();
-        Timer timer = new Timer();
-        timer.scheduleAtFixedRate(new TimerTask() {
-            @Override
-            public void run() {
-                if (autoLockRemainingMinutes == 1) {
-                    Logger.d("Autolock triggered");
-                    try {
-                        lockWallet();
-                    } catch (WalletNotInitialized ignored) {
-                        //we don't care about locking if it wasn't even inited
-                    }
-                    timer.cancel();
-                }
-                --autoLockRemainingMinutes;
 
+        Runnable onLockTriggered = () -> {
+            try {
+                frontendChannel.info("Autolock triggered");
+            } catch (Exception e) {
+                // Queue is full or no receiver found, ignore.
+            }
+            try {
+                lockWallet();
+            } catch (WalletNotInitialized ignored) {
+                //we don't care about locking if it wasn't even inited
```

Comment:
cannot there two try catch block merge into one? I can't find any reason to split them

Reply:
They can not be merged because if the first line throw exception then the `autoLock()` won't be executed. Anyway, you realized me that here is a different bug.  The `frontendChannel.info();` doesn't even throws any exception 😆 Fixed.